### ISSUE TITLE
Some intra-page links include the HTML file name

### ISSUE THREAD

**Issue by pawelszramowski** [doxygen/doxygen#10306]
**Describe the bug**
I observed that intra-page links (links to elements on the same page) generated by [`\ref`](https://www.doxygen.nl/manual/commands.html#cmdref) include the HTML file name.

For example, in [page1.html](https://www.doxygen.nl/manual/examples/page/html/page1.html) generated from [this example](https://www.doxygen.nl/manual/commands.html#cmdpage), intra-page links created by [`\tableofcontents`](https://www.doxygen.nl/manual/commands.html#cmdtableofcontents) are correct:

```html
<li class="level2"><a href="#subsection1">The first subsection</a></li><li class="level2"><a href="#subsection2">The second subsection</a></li>
```

whereas the links generated by `\ref` unnecessarily start with `page1.html`:

```html
This page contains the subsections <a class="el" href="page1.html#subsection1">The first subsection</a> and <a class="el" href="page1.html#subsection2">The second subsection</a>.
```

Intra-page links should not include the file name because:
1. It is not necessary for the links to work (links are relative to the current page).
2. It creates a dependency between the file name and content (if the file is renamed then the file content is no longer correct; not that it makes much sense to rename a single documentation page).
3. When the HTML page is printed to PDF in some web browsers, the links in the PDF file lead to elements within the original HTML file rather than within the PDF file. Mozilla Firefox is affected while Chromium-based Microsoft Edge works correctly.

**Expected behavior**
The HTML file name is not included in intra-page links, e.g. the aforementioned example is changed to:

```html
This page contains the subsections <a class="el" href="#subsection1">The first subsection</a> and <a class="el" href="#subsection2">The second subsection</a>.
```

**Screenshots**
None

**To Reproduce**
See bug description

**Version**
All?

**Stack trace**
N/A

**Additional context**
None

**Comment by albert-github** [comment]
For the test I used the current (1.9.8) doxygen version, an empty Doxyfile and the example as indicated: [example.tar.gz](https://github.com/doxygen/doxygen/files/12596608/example.tar.gz)

1. it is indeed not necessary in this case but due to the general setup (of e.g. the `\ref` command) it is, probably, hard to prevent / quite a bit of work. So this would be an enhancement
2. It indeed is a problem when renaming a file, but this does not make sense and when referring from other files these links would be broken anyway. 
3. This is something that, as far as I know, doxygen cannot influence. Printing to PDF from HTML pages (other than for direct printing on a printer, is not really advised).

**Comment by pawelszramowski** [comment]
2. The renaming may not even be explicit. Consider the following example:
    1. Open https://www.doxygen.nl/manual/examples/page/html/page1.html.
    2. Use the web browser's "Save Page As..." (Ctrl+S) option with the default file name.
    3. Firefox created the following file tree:
        ```
        Page Command A documentation page.htm
        Page Command A documentation page_files/
            doxygen.css
            doxygen.svg
            dynsections.js
            jquery.js
            menu.js
            menudata.js
            tabs.css
        ```
    The user may be fine with saving (and implicitly renaming) just a single page, you don't know that.
3. Of course, Doxygen cannot influence third-party programs, but it can make its output easier to process for them. By whom is printing to PDF from HTML pages not advised? It works just fine in my experience as long as you're aware of obvious gotchas, like links to other pages not working.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10338

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
